### PR TITLE
Implement, test, doc git descriptions for things

### PR DESCRIPTION
**Pull request by kshyatt**
Now you can ask for detailed descriptions of a repo's workdir or a commit-like object within it. cc @staticfloat.

### CODE REVIEW

File: base/libgit2/repository.jl [hunk 1]
```diff
@@ -239,6 +239,69 @@ function peel{T<:GitObject}(::Type{T}, obj::GitObject)
...
 
+"""
+    LibGit2.GitDescribeResult(commitish::GitObject; kwarg...)
+
+Produce a `GitDescribeResult` of the `commitish` `GitObject`, which
+contains detailed information about it based on the keyword argument:
+
+  * `options::DescribeOptions=DescribeOptions()`
+
+Equivalent to `git-describe <commitish>`.
```
Comment (ararslan) [inline base/libgit2/repository.jl]: I don't think there should be a hyphen between `git` and `describe`; `git describe` is the CLI AFAICT

**Comment by ararslan** [comment]
It might be nice to have the description interface be simply `LibGit2.describe(commitish, ...)`. That's a little more clear IMO than having to directly construct a `GitDescribeResult`.

**Comment by kshyatt** [comment]
I did it this way because it's a type and otherwise it lacks a constructor.

**Comment by ararslan** [comment]
`describe` could just be an alias for the constructor

File: test/libgit2.jl [hunk 2]
```diff
@@ -354,6 +354,7 @@ mktempdir() do dir
                     @test contains(showstr[4], "SHA:")
                     @test showstr[5] == "Message:"
                     @test showstr[6] == commit_msg1
+
```
Comment (tkelman) [inline test/libgit2.jl]: don't add empty lines as the first or last thing in a block

**Comment by tkelman** [comment]
we're in feature freeze, not the best time to be wrapping new api's

**Comment by kshyatt** [comment]
@staticfloat asked for it, we can wait until we're out of FF to merge if you want.

**Comment by kshyatt** [comment]
@tkelman I've rebased this against master. We're off feature freeze, right?

**Comment by kshyatt** [comment]
And bumping again. Can someone ( @ararslan ? @simonbyrne ? ) please review this?

**Review by simonbyrne — APPROVED** [review]
I made the small doc change suggested by @ararslan, LGTM once CI is done.

File: base/libgit2/repository.jl [hunk 3]
```diff
@@ -227,6 +227,69 @@ function peel(::Type{T}, obj::GitObject) where T<:GitObject
...
+    LibGit2.GitDescribeResult(repo::GitRepo; kwarg...)
+
+Produce a `GitDescribeResult` of the repository `repo`'s working directory,
+which can include all the commits and tags (or, for instance, HEAD only).
+The `GitDescribeResult` contains detailed information about the workdir based
+on the keyword argument:
+
+  * `options::DescribeOptions=DescribeOptions()`
+
+Equivalent to `git-describe`.
```
Comment (tkelman) [inline base/libgit2/repository.jl]: does the same apply here?

**Pull request merged**